# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1443259578 -7200
#      Sat Sep 26 11:26:18 2015 +0200
# Node ID 4479d73eac72ab0352a43aae4675d0cb4f4bd38f
# Parent  79eb0746b21f9063d97e8362217ce72a72aebbe2
Fix error when building annotation objects in gnuplot (bug #46035)

* __gnuplot_draw_figure__.m: ignore uicontextmenu objects.

diff --git a/scripts/plot/util/private/__gnuplot_draw_figure__.m b/scripts/plot/util/private/__gnuplot_draw_figure__.m
--- a/scripts/plot/util/private/__gnuplot_draw_figure__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_figure__.m
@@ -175,17 +175,17 @@ function __gnuplot_draw_figure__ (h, plo
                 ## Return axes "units" and "position" back to
                 ## their original values.
                 set (kids(i), "units", orig_axes_units);
                 set (kids(i), "position", orig_axes_position);
                 bg_is_set = false;
                 fg_is_set = false;
               end_unwind_protect
             endif
-          case "uimenu"
+          case {"uimenu", "uicontextmenu"}
             ## ignore uimenu objects
             kids(i) = [];
           otherwise
             error ("__gnuplot_draw_figure__: unknown object class, %s", type);
         endswitch
       endfor
       if (isempty (kids))
         fputs (plot_stream, "\nreset; clear;\n");
